whelk-core/src/main/groovy/whelk/search2/querytree/Link.java

```diff
@@ -1,17 +1,48 @@
 package whelk.search2.querytree;
 
-import whelk.search2.Disambiguate;
+import whelk.search2.VocabMappings;
 
-import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.Objects;
 
-public record Link(String iri, Object chip, String raw) implements Value {
-    public Link(String iri, Object chip) {
-        this(iri, chip, null);
+import static whelk.JsonLd.TYPE_KEY;
+import static whelk.JsonLd.asList;
+
+public final class Link extends Resource {
+    private final String iri;
+    private String raw;
+    private final Map<String, Object> thing = new LinkedHashMap<>();
+    private final Map<String, Object> chip = new LinkedHashMap<>();
+
+    public Link(String iri) {
+        this.iri = iri;
+    }
+
+    public Link(String iri, String raw) {
+        this.iri = iri;
+        this.raw = raw;
+    }
+
+    public Link(String iri, Map<String, Object> thing) {
+        this.iri = iri;
+        loadThing(thing);
+    }
+
+    public void loadChip(Map<String, Object> chip) {
+        this.chip.putAll(chip);
     }
 
-    public Link(String string) {
-        this(string, Collections.emptyMap());
+    public void loadThing(Map<String, Object> thing) {
+        this.thing.putAll(thing);
+    }
```

Comment:
Found this naming a bit confusing since "LinkLoader" also does "load".
`setChip`/`setThing`?